changelog/std_getopt_config_helplike.dd

```diff
@@ -0,0 +1,62 @@
+Add std.getopt.config.helpLike to std.getOpt module
+
+The new directive std.getopt.config.helpLike allows to use command-line arguments like
+additional 'help' arguments. The argument following this new directive is treated like
+'-h' or '--help'.
+
+A common use-case is an option to just output the (git-)version of a programm and then return
+with no error.
+
+In the current implementation of getOpt() it is needed to check the args array passed to
+main() to contain your special option '--version'.
+
+With this patch the option can be added to your getOpt() call. This increases readability
+of code and removed special case-handling of arguments before calling getOpt().
+
+So code like the following:
+---------
+int main(string[] args)
+{
+    /* Just test for '--version' in args here, before doing anything else */
+    auto printVersion = args.find!((a,b) => toLower(a) == b)("--version");
+    if (printVersion.length) {
+        printVersion();
+        return 0;
+    }
+
+    bool foo, bar, showVersion;
+    auto result = getopt(args,
+        std.getopt.config.required,
+        "foo", &foo,
+        "version", &showVersion, // Mask --version here or use option config.passThrough
+        "bar", &bar);
+    if (result.helpWanted) {
+        if (showVersion)
+            printVersion();  // Never called!
+        else
+            printHelp();
+        return 0;
+    }
+}
+---------
+
+can be changed to:
+---------
+int main(string[] args)
```

Comment:
I suggest adding a missing `import std.getopt`, definition for `printVersion`/`printHelp`, and `return 0;` to make it compilable.

Reply:
Good point.